### PR TITLE
Sockets auth management

### PR DESCRIPTION
**Pull request by shiryz**
Not ready for merge , waiting for #226 .
Added a different cookie to get the user id to the socket, sockets are now set with authorization to access the handshake.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -31,8 +40,11 @@ server.register([ inert, auth, ], (err) => {
...
 
   server.route(routes);
+
 });
 
-socket(server.listener);
+createSocket(server.listener);
 
-module.exports = server;
+module.exports = {
```
Comment (emilyb7) [inline server/server.js]: I know why we changed this originally, but it looks like `-module.exports = server;` would be ok now...

File: server/database/db_connection.js [hunk 2]
```diff
@@ -2,6 +2,8 @@ const pg = require('pg');
 const Pool = pg.Pool;
 const url = require('url');
 
+require('env2')('./config.env');
```
Comment (emilyb7) [inline server/database/db_connection.js]: I don't think we need this

File: server/routes/hello.js [hunk 3]
```diff
@@ -52,7 +52,8 @@ module.exports = {
               return reply(getUserDbErr + 'error getting user from database');
             }
             request.cookieAuth.set({ id: userId, });
-            reply.redirect('/');
+            reply.redirect('/').
+            state('new-user', fbUserId.toString());
```
Comment (des-des) [inline server/routes/hello.js]: this seems weird, are you sure you do not want to set this in the session you have aready created (see `request.cookieAuth.set({ id: userId, });` two lines above).

Also indentation

File: server/routes/logout.js [hunk 4]
```diff
@@ -4,7 +4,8 @@ module.exports = [
     method: 'GET',
     handler: (request, reply) => {
       request.cookieAuth.clear();
-      reply.redirect('/login');
+      reply.redirect('/login').
+      unstate('new-user');
```
Comment (des-des) [inline server/routes/logout.js]: Again

File: server/sockets.js [hunk 5]
```diff
@@ -1,12 +1,24 @@
...
-  io.on('connection', (socket) => {
-    console.log("new connection");
+  
+  io.set('authorization', (handshakeData, accept) => {
 
-    socket.on('newGoals', (data) => {
-      console.log(data);
-    });
+    if (handshakeData.headers.cookie) {
+      const cookie = cookieParser.parse(handshakeData.headers.cookie)['new-user'];
```
Comment (des-des) [inline server/sockets.js]: Okay I think you are really close.

I get that maybe your old system was not working with this. But it seems that you solution to this was to implement 2 authentication systems on the server. Is this correct? This does not seem like an optimal solution to your problem

File: server/sockets.js [hunk 6]
```diff
@@ -9,7 +10,11 @@ const createSocket = (listener) => {
 
     if (handshakeData.headers.cookie) {
       const cookie = cookieParser.parse(handshakeData.headers.cookie)['grow-user'];
-      console.log('cookie ', cookie);
+      iron.unseal(cookie, process.env.COOKIE_PASSWORD, iron.defaults, (err, unsealed) => {
+        console.log('cookie: ',unsealed);
```
Comment (des-des) [inline server/sockets.js]: :fire:

File: server/sockets.js [hunk 7]
```diff
@@ -9,7 +10,11 @@ const createSocket = (listener) => {
...
+        // unsealed has the same content as obj
```
Comment (des-des) [inline server/sockets.js]: do we want this comment?

File: server/sockets.js [hunk 8]
```diff
@@ -9,7 +10,11 @@ const createSocket = (listener) => {
...
+      });
+      // console.log('cookie ', cookie);
```
Comment (des-des) [inline server/sockets.js]: :fire:

File: server/sockets.js [hunk 9]
```diff
@@ -1,12 +1,17 @@
...
-      console.log(data);
-    });
+  io.use((socket, next) => {
+    if(socket.request.headers.cookie){
+      next();
+    }
+    next(new Error('Authentication error'));
   });
+
+  io.on('connection', socketManager);
```
Comment (des-des) [inline server/sockets.js]: Socket manager does not do anything...

Comment (des-des) [inline server/sockets.js]: No return, no side effects

**Pull request merged**